test/extended/images/hardprune.go

```diff
@@ -475,22 +475,84 @@ func ConfigureRegistry(oc *exutil.CLI, desiredState RegistryConfiguration) error
 
 	err = oc.Run("env").Args(append([]string{"dc/docker-registry"}, envOverrides...)...).Execute()
 	if err != nil {
-		return fmt.Errorf("failed to update registry's environment with %s: %v", &waitForVersion, err)
+		return false, fmt.Errorf("failed to update registry's environment with %s: %v", &waitForVersion, err)
 	}
-	return exutil.WaitForDeploymentConfig(
+	err = exutil.WaitForDeploymentConfig(
 		oc.AdminKubeClient(),
 		oc.AdminAppsClient().Apps(),
 		metav1.NamespaceDefault,
 		"docker-registry",
 		waitForVersion,
 		oc)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func RedeployRegistry(oc *exutil.CLI) (bool, error) {
+	defer func(ns string) { oc.SetNamespace(ns) }(oc.Namespace())
+
+	oc = oc.SetNamespace(metav1.NamespaceDefault).AsAdmin()
+
+	env, err := oc.Run("env").Args("dc/docker-registry", "--list").Output()
+	if err != nil {
+		return false, err
+	}
+
+	num := 0
+
+	for _, envVar := range strings.Split(env, "\n") {
+		if len(envVar) == 0 {
+			continue
+		}
+
+		envKV := strings.SplitN(envVar, "=", 2)
+		if len(envKV) == 0 || envKV[0] != "REDEPLOY_REGISTRY" {
+			continue
+		}
+
+		i, err := strconv.Atoi(envKV[1])
+		if err != nil {
+			continue
+		}
+
+		num = i
+	}
+
+	dc, err := oc.AppsClient().Apps().DeploymentConfigs(metav1.NamespaceDefault).Get("docker-registry", metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	// log docker-registry pod output before re-deploying
+	waitForVersion := dc.Status.LatestVersion + 1
+	if err = LogRegistryPod(oc); err != nil {
+		fmt.Fprintf(g.GinkgoWriter, "failed to log registry pod: %v\n", err)
+	}
+
+	err = oc.Run("env").Args(append([]string{"dc/docker-registry"}, fmt.Sprintf("REDEPLOY_REGISTRY=%d", num+1))...).Execute()
```

Comment:
any reason you didn't just do an "oc rollout latest" to force the redeployment, instead of all this env mainipulation?

Reply:
OK. I will do like you say.

Reply:
Now I have an answer for you. Because if you execute the `oc rollout latest`, you will get: "error: #6 is already in progress (Running)."

Reply:
If there is already another deployment happening when you start updating the env vars then you're going to have another problem:  As soon as that deployment finishes, your env var update is going to trigger another deployment, potentially interrupting your test.  (Because your "waitfordeployment" will wait for the first deployment, not the second, i believe)

How about checking the DC to see if a deployment is in progress, before doing the oc rollout?

Reply:
@bparees I added the `--again=true` option and it seems like everything is working now. That's better ?